jsaddle/js/jsaddle-core.js

```diff
@@ -123,9 +125,33 @@ function jsaddle(global, sendRsp, processSyncCommand) {
   var wrapVal = function(val) {
     return wrapValWithDefault(val);
   };
+  var doSendRsp = function () {
+    if (responses.length > 0) {
+      sendRsp(responses);
```

Comment:
assuming this isn't immediate, this is racy and could cause multiple doSendRsp to got out at once. We should do `var responses_ = responses; responses = []` first so that we don't send multiple responses.